toplevel/opttopmain.ml

```diff
@@ -20,6 +20,26 @@ let usage =
 
 let preload_objects = ref []
 
+(* Position plus length of last expand argument *)
+let expand_offset: (int * int) option ref = ref None
```

Comment:
I re-reviewed the code and it seems correct to me, but a bit complicated. We could add more comments to explain, but I wonder if there is not a simpler way. For example, why would it not be enough to only keep the limit expanded arguments (what you call "position + length")? I'm not sure what the position is for; it would be useful if positions could decrease over time, but is that possible?